gen_cert.py

```diff
@@ -233,12 +236,16 @@ def interstitial_factory():
             log.critical("Unable to lookup long names for course {0}".format(course_id))
             raise
 
-        # split the org and course from the course_id
-        # if COURSE or ORG is set in the configuration
-        # dictionary, use that instead
-        tmp_org, tmp_course, tmp_run = course_id.split('/')
-        self.course = cert_data.get('COURSE', tmp_course)
-        self.org = cert_data.get('ORG', tmp_org)
+        # if COURSE or ORG is set in the configuration attempt to parse using
+        # new style course_keys and fallback to using from_deprecated_string
+        # upon failure
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
```

Comment:
Please use `CourseKey.from_string()` -- that should handle both the new style locators and the old slash separated course key cases.